### ISSUE TITLE
Checker shouldn't modify the AST

### ISSUE THREAD

**Issue by danieldaeschle** [vlang/v#8059]
Something like this shouldn't be in checker. It should be in cgen.
![image](https://user-images.githubusercontent.com/30751516/104299167-29b5c180-54c5-11eb-8ee7-308ec061fd26.png)
![image](https://user-images.githubusercontent.com/30751516/104299187-31756600-54c5-11eb-85be-573823203a80.png)
![image](https://user-images.githubusercontent.com/30751516/104299501-97fa8400-54c5-11eb-8caf-d3ae1984fb4f.png)

**Comment by ArtemkaKun** [comment]
Hi @danieldaeschle
Is this still actual? The code of checker changed since then + I don't really understand what should be done in cgen. Access to `.stmts`?

**Comment by danieldaeschle** [comment]
I don't know anymore. Checker shouldn't modify the AST imho. That was the point

**Comment by JalonSolov** [comment]
I disagree.  cgen should never modify anything, it should only generate C code.

We already have too many things being done in cgen, which means they have to be duplicated in jsgen, and nativegen, and every new gen that gets created.

The backends should only ever generate code from the AST - they should never modify the AST.

**Comment by ArtemkaKun** [comment]
I can understand the point about the checker too since the checker should only "check", I think. Maybe there should be some kind of step "modify AST based on checker output"?

**Comment by JalonSolov** [comment]
There is a `transform` step now.  The modification could be done there.  I just don't want to see any more things moved to cgen.

**Comment by Larpon** [comment]
Just for completeness; the [transformer step is here](https://github.com/vlang/v/blob/master/vlib/v/transformer).

**Comment by medvednikov** [comment]
Fixed in V2.
